src/integration/java/io/pinecone/integration/dataPlane/UpdateFetchAndQueryPodTest.java

```diff
@@ -6,74 +6,71 @@
 import io.pinecone.unsigned_indices_model.ScoredVectorWithUnsignedIndices;
+import io.pinecone.unsigned_indices_model.SparseValuesWithUnsignedIndices;
+import io.pinecone.unsigned_indices_model.VectorWithUnsignedIndices;
 import org.junit.jupiter.api.AfterAll;
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
-import org.openapitools.client.model.IndexModelSpec;
 
 import java.io.IOException;
 import java.util.*;
 import java.util.concurrent.ExecutionException;
 
 import static io.pinecone.helpers.AssertRetry.assertWithRetry;
 import static io.pinecone.helpers.BuildUpsertRequest.*;
-import static io.pinecone.helpers.IndexManager.*;
 import static org.junit.jupiter.api.Assertions.*;
 
-public class UpdateFetchAndQueryPodTest {
+public class  UpdateFetchAndQueryPodTest {
+    private static final TestIndexResourcesManager indexManager = TestIndexResourcesManager.getInstance();
     private static Index index;
     private static AsyncIndex asyncIndex;
-    private static String namespace;
+    private static final String namespace = RandomStringBuilder.build("ns", 8);
     private static List<String> upsertIds;
-    private static List<List<Float>> sparseValuesList;
-    private static final int dimension = 3;
+    private static List<String> sparseUpsertIds;
+    private static List<Float> sparseValuesList;
+    private static int dimension;
 
     @BeforeAll
     public static void setUp() throws IOException, InterruptedException {
-        String apiKey = System.getenv("PINECONE_API_KEY");
-        String indexType = IndexModelSpec.SERIALIZED_NAME_POD;
-        Pinecone pinecone = new Pinecone.Builder(apiKey).build();
+        Pinecone pinecone = new Pinecone.Builder(System.getenv("PINECONE_API_KEY")).build();
 
-        String indexName = findIndexWithDimensionAndType(pinecone, dimension, indexType);
-        if (indexName.isEmpty()) indexName = createNewIndex(pinecone, dimension, indexType, true);
+        String indexName = indexManager.getPodIndexName();
+        dimension = indexManager.getDimension();
         index = pinecone.getIndexConnection(indexName);
         asyncIndex = pinecone.getAsyncIndexConnection(indexName);
 
         // Upsert vectors only once
         int numOfVectors = 3;
         int numOfSparseVectors = 2;
-        namespace = RandomStringBuilder.build("ns", 8);
         upsertIds = getIdsList(numOfVectors);
-        List<List<Long>> sparseIndicesList = getSparseIndicesList(numOfSparseVectors, dimension);
-        sparseValuesList = getValuesListLists(numOfSparseVectors, dimension);
-
-        // upsert vectors with required + optional parameters
-        int sparseVectorCount = 0;
-        for (int i = sparseVectorCount; i < numOfSparseVectors; i++) {
-            index.upsert(upsertIds.get(i),
+        sparseUpsertIds = getIdsList(numOfSparseVectors);
+        ArrayList<Long> sparseIndices = generateSparseIndicesByDimension(dimension);
```

Comment:
Previously we were for-looping and calling `upsert` each time. Rather than doing that, I thought it'd be better to construct a `List<VectorWithUnsignedIndices` and build that, then call `upsert` once. I've done that for a bunch of the data plane tests.